ophyd/controls/signal.py

```diff
@@ -285,6 +285,13 @@ def _connected(self, pvname=None, conn=None, pv=None, **kwargs):
             self._ses_logger.debug(msg)
 
     @property
+    def connected(self):
+        if self._write_pv is None:
+            return self._read_pv.connected
+        else:
+            return self._read_pv.connected and self._write_pv.connected
```

Comment:
I personally prefer it the way @dchabot wrote it. I find that more readable. Giant (multi-case) if/elif/elif/.../else blocks are an anti-pattern in Python -- maybe that is what @tacaswell was talking about?